plugins/foreman_ansible/2.x/index.md

```diff
@@ -19,42 +19,124 @@ You can define a list of roles to enforce, or just run arbitrary playbooks on th
+</p>
+
+{% highlight bash %}
+[defaults]
+callback_whitelist = foreman
+{% endhighlight %}
+
+<p>
 and the callback will be enabled after that.
-
-
-For Ansible versions older than 2.2:
-
-* Change your `/etc/ansible/ansible.cfg` to contain the following options
-
-      callback_plugins = ~/.ansible/plugins/callback_plugins/
-      bin_ansible_callbacks = True
-
-* And run this line to download the callback file in the appropriate location
-
-      wget https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/foreman.py -O ~/.ansible/plugins/callback_plugins/foreman.py
-
-* Alternatively, you can find the source for the [Ansible callback here](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/foreman.py) and deploy it in your directory of choice.
+</p>
+
+<p>
+Of course, the callback cannot know automatically where is Foreman located and which certificates to use in its requests.
+</p>
+</div>
+
+<div class="ansible_version ansible_version_25" style="display:none;">
+<p>
+To configure it, add a section <code>[callback_foreman]</code> at the end of <code>/etc/ansible/ansible.cfg</code> like this:
+</p>
+
+{% highlight bash %}
+[callback_foreman]
+url = 'https://example.com'
+ssl_cert = /etc/foreman-proxy/ssl_cert.pem
+ssl_key = /etc/foreman-proxy/ssl_key.pem
+verify_certs = /etc/foreman-proxy/ssl_ca.pem
+{% endhighlight %}
+</div>
+
+<div class="ansible_version ansible_version_21" style="display:none;">
+<p>
+We don't really test these versions, however if you need to use an old version of Ansible, you could do the following:
+</p>
+
+<ul>
+  <li>
+  Change your <code>/etc/ansible/ansible.cfg</code> to contain the following options:
+  </li>
+
+{% highlight bash %}
+callback_plugins = ~/.ansible/plugins/callback_plugins/
+bin_ansible_callbacks = True
+{% endhighlight %}
+
+  <li>
+  And run this line to download the callback file in the appropriate location:
+  </li>
+
+{% highlight bash %}
+wget https://github.com/ansible/ansible/blob/stable-2.2/lib/ansible/plugins/callback/foreman.py -O ~/.ansible/plugins/callback_plugins/foreman.py
+{% endhighlight %}
+</ul>
+</div>
+
+<div class="ansible_version ansible_version_22 ansible_version_21" style="display:none;">
+
+<p>
+You need to configure it via the following environment variables (the variables need to be available to the user running the callback. Usually <code>foreman-proxy</code>):
+</p>
+
+<ul>
+  <li>
+    FOREMAN_URL: the URL of your Foreman installation (default "http://localhost:3000")
+  </li>
+  <li>
+    FOREMAN_SSL_CERT: The public key when using SSL client certificates (default "/etc/foreman/client_cert.pem")
+  </li>
+  <li>
+    FOREMAN_SSL_KEY: The private key when using SSL client certificates (default  "/etc/foreman/client_key.pem")
+  </li>
+  <li>
+    FOREMAN_SSL_VERIFY: wether to verify SSL certificates. Use *False*
+to disable certificate checks. You can also set it to CA bundle (default is "True").
```

Comment:
I think you could set it to `True` for the CA bundle or a path to the CA file for a custom CA (as it often is).

Reply:
correct